### PR TITLE
GH-936: Exposed low-level Git exec function.

### LINKED ISSUE #936: [git] Git API should expose a low-level `exec` function
Git API should expose a low-level `exec` function so that Theia extension developers could implement their own functions which are not yet supported by Theia's Git API.

Later one could open a PR in Theia with these additional, custom Git commands. If those seem to be useful for general, public usage, we could merge those into our API.

### PR DESCRIPTION
**Pull request by kittaakos**
Closes #936.

Signed-off-by: Akos Kitta <kittaakos@gmail.com>

### CODE REVIEW

**Comment by kittaakos** [comment]
Something is wrong with the repository watcher in the tests. It does not release the lock from a repository after executing a failed command on Windows.

**Comment by kittaakos** [comment]
> It does not release the lock from a repository

I cannot reproduce it locally, but I got the #933 problem.

File: packages/git/src/common/git.ts [hunk 1]
```diff
@@ -562,4 +637,39 @@ export namespace GitUtils {
         return false;
     }
 
+    /**
+     * Maps the raw status text from Git to a Git file status enumeration.
+     */
+    export function mapStatus(rawStatus: string): GitFileStatus {
```
Comment (hexa00) [inline packages/git/src/common/git.ts]: is that used by anything ?

Comment (kittaakos) [reply packages/git/src/common/git.ts]: No, not yet. Just like the `exec` function, but we would like to provide flexibility for future extension developers with the BETA release.

Comment (svenefftinge) [reply packages/git/src/common/git.ts]: It is defined to be used by potential extension developers but it should be tested at least.
@kittaakos could you add one?

**Comment by kittaakos** [comment]
@svenefftinge, are we good with this? I had to apply some changes after you did the review and approved it. Thanks!

**Comment by kittaakos** [comment]
> could you add one?

See: https://github.com/theia-ide/theia/pull/940/commits/b57bae6a6b297751a2dbb22b14482ef1c7e0a106#diff-372b43f7c1ace74b93cb267b0747312cR319

**Pull request merged**